src/main/java/seedu/duke/utils/Parser.java

```diff
@@ -0,0 +1,336 @@
+            switch (operation) {
+            case BYE:
+                toReturn = taskList.exit();
+                break;
+            case LIST:
+                toReturn = taskList.listAllTasks();
+                break;
+            case DELETE:
+                toReturn = delete(userInput, taskList);
+                break;
+            case MARK:
+                toReturn = mark(userInput, taskList);
+                break;
+            case UNMARK:
+                toReturn = unmark(userInput, taskList);
+                break;
+            case TODO:
+                toReturn = todo(userInput, taskList);
+                break;
+            case DEADLINE:
+                toReturn = deadline(userInput, taskList);
+                break;
+            case EVENT:
+                toReturn = event(userInput, taskList);
+                break;
+            case CHECKDATE:
+                toReturn = checkDate(userInput, taskList);
+                break;
+            case FIND:
+                toReturn = find(userInput, taskList);
+                break;
+            default:
+                throw new InvalidCommandException();
+            }
+            storage.writeFile(taskList);
+            return toReturn;
+        } catch (TaskException | InvalidCommandException | EmptyDescriptionException
+                 | NotIntegerException | MissingKeywordException e) {
+//            System.out.println(e.getMessage());
+            return e.getMessage();
+        }
+    }
+
+    /**
+     * calls the deleteTask function in TaskList
+     *
+     * @param userInput user input
+     * @param taskList taskList to operate on
+     * @throws TaskException exception thrown
+     * @throws NotIntegerException exception thrown
+     */
+    public String delete(String userInput, TaskList taskList) throws TaskException, NotIntegerException {
+        String[] parts = userInput.split(" ", 2);
+        if (!isInteger(parts[1])) {
+            throw new NotIntegerException();
+        }
+        int taskIndex = Integer.parseInt(parts[1]);
+        // error of out of bounds handled in TaskList itself
+        // since the size is better/ easier to get in TaskList class
+        return taskList.deleteTask(taskIndex);
+    }
+
+    /**
+     * calls the mark function in taskList
+     *
+     * @param userInput user input
+     * @param taskList tasklist
+     * @throws EmptyDescriptionException exception
+     * @throws NotIntegerException exception
+     * @throws TaskException exception
+     */
+    public String mark(String userInput, TaskList taskList) throws EmptyDescriptionException,
+            NotIntegerException, TaskException {
+        String details = extractNoKeywordsDetails(userInput);
+        if (!isInteger(details)) {
+            throw new NotIntegerException();
+        }
+        int taskIndex = Integer.parseInt(details);
+        return taskList.mark(taskIndex);
+    }
+
+    /**
+     * calls the unmark function
```

Comment:
For javadoc comments, can try to capitalise the first letter of sentence so it looks nicer